hibernate-core/src/main/java/org/hibernate/engine/jdbc/spi/SqlStatementLogger.java

```diff
@@ -134,34 +135,34 @@ public void logStatement(String statement, Formatter formatter) {
 	 * Log a slow SQL query
 	 *
 	 * @param statement SQL statement.
-	 * @param startTime Start time in milliseconds.
+	 * @param startTimeNanos Start time in nanoseconds.
 	 */
-	public void logSlowQuery(Statement statement, long startTime) {
+	public void logSlowQuery(Statement statement, long startTimeNanos) {
 		if ( logSlowQuery < 1 ) {
 			return;
 		}
-		logSlowQuery( statement.toString(), startTime );
+		logSlowQuery( statement.toString(), startTimeNanos );
 	}
 
 	/**
 	 * Log a slow SQL query
 	 *
 	 * @param sql The SQL query.
-	 * @param startTime Start time in milliseconds.
+	 * @param startTimeNanos Start time in nanoseconds.
 	 */
 	@AllowSysOut
-	public void logSlowQuery(String sql, long startTime) {
+	public void logSlowQuery(String sql, long startTimeNanos) {
 		if ( logSlowQuery < 1 ) {
 			return;
 		}
-		assert startTime > 0 : "startTime is invalid!";
-
-		long spent = System.currentTimeMillis() - startTime;
+		if ( startTimeNanos <= 0 ) {
+			throw new IllegalArgumentException( "startTimeNanos should be greater than 0!" );
```

Comment:
I would suggest providing more clue about  the exception like
```
throw new IllegalArgumentException( "startTimeNanos [" + startTimeNanos + "] should be greater than 0!" );
```
to help debugger a little bit.